lib/puppet/provider/service/smf.rb

```diff
@@ -129,6 +129,14 @@ def status
       return
     end
 
+    # Check to see if the service exists
+    cmd = Array[command(:svccfg), "select", @resource[:name]]
+    output = Puppet::Util::Execution.execute(
+        cmd, :combine => true, :failonfail => false)
+    if $CHILD_STATUS.exitstatus != 0
+      info output
+    end
```

Comment:
Puppet.debug == debug :: Puppet.info == info; I can fully specify the call if that is the style.

I don't think I can reference the method without the rescue which seems like extra work. as opposed to passing fail_on_fail => false